File: Practice problem/Find prime number/Find prime number.cpp
# 연습문제 소수 찾기 Level 1

#include <cmath>

using namespace std;

int solution(int n) {
    int primelist[500000] = {0};
    int answer = 0;
    bool is_prime = true;
    int root = 0;
    for(int i = 2; i < n+1 ; i++){
        is_prime = true;
        root = ceil(sqrt(i));
        for(int j = 0; primelist[j] != 0 && primelist[j] <= root ; j++){
            if (i % primelist[j] == 0) {
                is_prime = false;
                break;
            }
        }
        if (is_prime) {
            primelist[answer] = i;
            answer++;
        }
    }
    return answer;
}

File: Brute-force Search/Find Prime Number.py
# 완전탐색 소수 찾기 level 2

import math
from itertools import permutations

def prime_check(number):
    if(number < 2):
        return False
    check = math.sqrt(number)
    for i in range(2, int(check)+1):
        if( number % i == 0):
            return False
    return True

def solution(numbers):
    answer = []
    for i in range(1, len(numbers)+1):
        numset = list(set(map("".join, permutations(list(numbers), i))))
        for j in range(len(numset)):
            if(prime_check(int(numset[j]))):
                answer.append(int(numset[j]))
    return len(set(answer))
